### PR TITLE
Add Launch on Startup - Electron-25

### PR DESCRIPTION
**Pull request by rbarros-daitan**
Implements ELECTRON-25 and also prepare the basics for ELECTRON-40 - Minimize On Close.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -86,6 +86,7 @@
     "electron-squirrel-startup": "^1.0.0",
     "keymirror": "0.1.1",
     "electron-context-menu": "^0.8.0",
+    "windows-shortcuts": "^0.1.4",
```
Comment (lneir) [inline package.json]: i am not sure about using this library. it is bringing in some external shortcut.exe program.  i don't think we can trust that.  

other options
- build a native node module like: https://github.com/nadavbar/node-win-shortcut
- find another program that comes installed with 7 and greater that does same thing as shortcut.exe

i prefer the native node module myself.

File: installer/Symphony-x64.aip [hunk 2]
```diff
@@ -63,17 +63,16 @@
...
   </COMPONENT>
   <COMPONENT cid="caphyon.advinst.msicomp.MsiFeatsComponent">
     <ROW Feature="D564007E3BBE4F85950A09B470A7CA65" Title="Visual C++ Redistributable for Visual Studio 2013 x86" Description="Visual C++ Redistributable for Visual Studio 2013 x86" Display="3" Level="1" Attributes="0" Components="AI_CustomARPName"/>
-    <ROW Feature="MainFeature" Title="MainFeature" Description="Description" Display="1" Level="1" Directory_="APPDIR" Attributes="0" Components="AI_CustomARPName PodUrl ProductInformation Symphony Symphony.config Symphony.exe am.pak appupdate.yml blink_image_resources_200_percent.pak d3dcompiler_47.dll ffmpeg.dll libEGL.dll libGLESv2.dll node.dll xinput1_3.dll"/>
+    <ROW Feature="MainFeature" Title="MainFeature" Description="Description" Display="1" Level="1" Directory_="APPDIR" Attributes="0" Components="AI_CustomARPName PodUrl ProductInformation Symphony Symphony.config Symphony.exe am.pak appupdate.yml blink_image_resources_200_percent.pak d3dcompiler_47.dll ffmpeg.dll libEGL.dll libGLESv2.dll node.dll"/>
```
Comment (lneir) [inline installer/Symphony-x64.aip]: please create a separate PR that fixes .aip file for electron 1.6.7.  thanks.

File: package.json (hunk 1 above)
Comment (rbarros-daitan) [inline package.json]: Hi Lynn. 
node-auto-launch seems to be much more elegant than windows-shortcuts. I´m using this, will commit the change in some minutes.

File: installer/Symphony-x64.aip (hunk 2 above)
Comment (rbarros-daitan) [inline installer/Symphony-x64.aip]: ok.

File: installer/Symphony-x64.aip [hunk 3]
```diff
@@ -63,16 +64,17 @@
     <ROW Component="libEGL.dll" ComponentId="{8EEC76AB-3601-4D11-B13E-32EC2A38C539}" Directory_="APPDIR" Attributes="256" KeyPath="libEGL.dll"/>
     <ROW Component="libGLESv2.dll" ComponentId="{0E8B8B21-B4C0-45C9-95D3-637FD93A4EC0}" Directory_="APPDIR" Attributes="256" KeyPath="libGLESv2.dll"/>
     <ROW Component="node.dll" ComponentId="{C0972355-339E-438C-94A3-74174DE4C6B6}" Directory_="APPDIR" Attributes="256" KeyPath="node.dll"/>
+    <ROW Component="xinput1_3.dll" ComponentId="{0B086863-0C7F-4722-B986-FF90379D51B2}" Directory_="APPDIR" Attributes="256" KeyPath="xinput1_3.dll"/>
```
Comment (lneir) [inline installer/Symphony-x64.aip]: already fixed this in separate PR.  rebase and shouldn't be needed.

File: installer/Symphony-x64.aip [hunk 4]
```diff
@@ -63,16 +64,17 @@
...
   </COMPONENT>
   <COMPONENT cid="caphyon.advinst.msicomp.MsiFeatsComponent">
     <ROW Feature="D564007E3BBE4F85950A09B470A7CA65" Title="Visual C++ Redistributable for Visual Studio 2013 x86" Description="Visual C++ Redistributable for Visual Studio 2013 x86" Display="3" Level="1" Attributes="0" Components="AI_CustomARPName"/>
-    <ROW Feature="MainFeature" Title="MainFeature" Description="Description" Display="1" Level="1" Directory_="APPDIR" Attributes="0" Components="AI_CustomARPName PodUrl ProductInformation Symphony Symphony.config Symphony.exe am.pak appupdate.yml blink_image_resources_200_percent.pak d3dcompiler_47.dll ffmpeg.dll libEGL.dll libGLESv2.dll node.dll"/>
+    <ROW Feature="MainFeature" Title="MainFeature" Description="Description" Display="1" Level="1" Directory_="APPDIR" Attributes="0" Components="AI_CustomARPName PodUrl ProductInformation Symphony Symphony.config Symphony.exe Symphony_1 am.pak appupdate.yml blink_image_resources_200_percent.pak d3dcompiler_47.dll ffmpeg.dll libEGL.dll libGLESv2.dll node.dll xinput1_3.dll"/>
```
Comment (lneir) [inline installer/Symphony-x64.aip]: ditto above comment.

File: js/menus/menuTemplate.js [hunk 5]
```diff
@@ -2,14 +2,20 @@
...
-const ws = require('windows-shortcuts');
-const fs = require('fs');
-const path = require('path');
+const AutoLaunch = require('auto-launch');
 
 var minimizeOnClose = false;
 var launchOnStartup = false;
 
 setCheckboxValues();
+console.log(process.execPath);
```
Comment (lneir) [inline js/menus/menuTemplate.js]: remove consoe.log

File: js/menus/menuTemplate.js [hunk 6]
```diff
@@ -1,6 +1,21 @@
...
 
 const electron = require('electron');
+const { getConfigField, updateConfigField } = require('../config.js');
+const AutoLaunch = require('auto-launch');
+
+var minimizeOnClose = false;
+var launchOnStartup = false;
+
+setCheckboxValues();
+console.log(process.execPath);
```
Comment (lneir) [inline js/menus/menuTemplate.js]: remove console.log

File: js/menus/menuTemplate.js [hunk 7]
```diff
@@ -156,9 +171,73 @@ function getTemplate(app) {
...
+                    updateConfigField('launchOnStartup', launchOnStartup);
+                }
+            }
+        )
     }
 
+    // Window menu -> minimizeOnClose.
+    // ToDo: Add behavior on Close.
+    var index = 2;
+    if (process.platform === 'darwin' && template[0].label !== app.getName()){
```
Comment (lneir) [inline js/menus/menuTemplate.js]: already have util isMac

Comment (lneir) [inline js/menus/menuTemplate.js]: why check for mac here?

File: js/menus/menuTemplate.js [hunk 8]
```diff
@@ -156,9 +171,73 @@ function getTemplate(app) {
...
+        index = 3;
+    }
+    template[index].submenu.push(
```
Comment (lneir) [inline js/menus/menuTemplate.js]: is the minimize feature on close completed?  mac currently has the feature by default, see windowMgr.js.   this flag should be used there to enable/disable close.

File: js/menus/menuTemplate.js [hunk 9]
```diff
@@ -156,9 +171,73 @@ function getTemplate(app) {
                 role: 'front'
             }
         ]
+    } else {
+      // Window menu when Windows.
+        template[2].submenu.push(
+            {
+                label: 'Auto Launch on Windows Startup',
```
Comment (lneir) [inline js/menus/menuTemplate.js]: why Windows here.  Doesn't this feature work on both mac and windows when using AutoLaunch lib?  has this been tested?

File: js/menus/menuTemplate.js [hunk 10]
```diff
@@ -156,9 +171,73 @@ function getTemplate(app) {
...
+      // Window menu when Windows.
+        template[2].submenu.push(
+            {
+                label: 'Auto Launch on Windows Startup', 
+                type: 'checkbox', 
+                checked: launchOnStartup,
+                click: function (item) {
+                    if (item.checked){
+
+                        symphonyAutoLauncher.isEnabled()
```
Comment (lneir) [inline js/menus/menuTemplate.js]: i need to confirm with product but believe expectation here is that auto launch is enabled after installing the app.  we might have to invoke the app during install process with special flag that sets launch values then exits immediately.

**Pull request closed without merging**